addons/stock/migrations/8.0.1.1/post-migration.py

```diff
@@ -557,8 +557,8 @@ def migrate_product_supply_method(cr, registry):
 
     if mto_route_id:
         product_ids = []
-        cr.execute("SELECT id FROM product_template WHERE %s = %%s" % (
-            procure_method_legacy, ('make_to_order',)))
+        cr.execute("SELECT id FROM product_template WHERE %s = '%s'" % (
+            procure_method_legacy, 'make_to_order'))
```

Comment:
What the previous attempt meant to do, was to use Psycopg2's built-in, safe variable substitution where possible which is a (implicit) convention in the OpenUpgrade project. This leads to hybrid statements when substituting both values and column or table names because the latter cannot be substituted by Psycopg2. According to this convention, the fix would be something like

```
cr.execute("SELECT id FROM product_template WHERE %s = %%s" % (
    procure_method_legacy), ('make_to_order',))
```